### PR TITLE
docs: update networking description

### PR DESCRIPTION
**Pull request by egernst**
First, most people don't care about CNM. Move that out of main doc.

Second, tc-filter is the default. Let's add a bit more background on
our usage of tc-filter (and clarify why we use this instead of macvtap).

Fixes: #692

Signed-off-by: Eric Ernst <eric.g.ernst@gmail.com>

### CODE REVIEW

File: design/architecture.md [hunk 1]
```diff
@@ -386,65 +386,31 @@ In order to do so, container engines will usually add one end of a virtual
...
-4.  `CreateEndPoint`
-
-5.  `CreateContainer`
-
-6.  Create `config.json`
-
-7.  Create PID and network namespace
+With TC filter in place, a redirection is created between the container network and the
+virtual machine. As an example, the CNI may create a device, eth0, in the container's network
+namespace, which is a veth device. Kata Containers will create a tap device for the VM, kata_tap0,
```
Comment (jodh-intel) [inline design/architecture.md:400]: - `kata_tap0` to keep the spell-checker happy.
- veth -> VETH as the spell checker undestands that.

File: design/architecture.md [hunk 2]
```diff
@@ -386,65 +386,31 @@ In order to do so, container engines will usually add one end of a virtual
...
-
-4.  `CreateEndPoint`
-
-5.  `CreateContainer`
-
-6.  Create `config.json`
-
-7.  Create PID and network namespace
+With TC filter in place, a redirection is created between the container network and the
+virtual machine. As an example, the CNI may create a device, eth0, in the container's network
```
Comment (jodh-intel) [inline design/architecture.md:399]: `eth0`

File: design/architecture.md [hunk 3]
```diff
@@ -386,65 +386,31 @@ In order to do so, container engines will usually add one end of a virtual
...
+namespace, which is a veth device. Kata Containers will create a tap device for the VM, kata_tap0,
+and setup a TC redirection filter to mirror traffic from eth0's ingress to kata_tap0's egress,
+and a second to mirror traffic from kata_tap0's ingress to eth0's egress.
```
Comment (jodh-intel) [inline design/architecture.md:402]: `kata_tap0` and `eth0` everywhere.

File: design/architecture.md [hunk 4]
```diff
@@ -386,65 +386,31 @@ In order to do so, container engines will usually add one end of a virtual
...
 
-8.  `ProcessExternalKey`
+Kata maintains support for MACVTAP, which was an earlier implementation in the kata-runtime. TC-filter
+is the default because it allows for simpler configuration, better CNI plugin compatibility, and performance
+on par with MACVTAP.
 
-9.  `JoinEndPoint`
+Kata has deprecated support for bridge due to lacking performance relative to tc-filter and macvtap.
```
Comment (jodh-intel) [inline design/architecture.md:408]: To keep the spell checker happy:

- s/tc-filter/TC-filter/
- s/macvtap/MACVTAP/

**Comment by jodh-intel** [comment]
Oh - also Travis has detected a document referencing issue:

```
ERROR: Document design/cnm.md is not referenced
```

**Comment by egernst** [comment]
thanks for the quick review @jodh-intel 
Updated.

**Comment by cmaf** [comment]
Forward port is https://github.com/kata-containers/kata-containers/pull/798

**Pull request merged**